### PR TITLE
refactor log analysis

### PR DESCRIPTION
**Pull request by gurtanvi**
add .gitignore
refactor log_analysis

### CODE REVIEW

File: log.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        'name': get_operation_name(line),
+        'type': get_operation_type(line),
+        'duration': get_operation_duration(line),
+    }
+    return operation
+
+
+
+
+#df_logfile = create_log_df('/home/gurleenkaur/Programming/log_analysis/data/logs.log')
```
Comment (madhavhugar) [inline log.py]: uncomment this line 

```python
df_logfile = create_log_df('/home/gurleenkaur/Programming/log_analysis/data/logs.log')
```
`df_logfile` would be a dataframe with 3 columns `name`, `type`, and `duration`

you are headed in the right direction. Once you have the dataframe apply some pandas magic to get the expected results 🧙

File: log.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        'type': get_operation_type(line),
+        'duration': get_operation_duration(line),
+    }
+    return operation
+
+
+
+
+df_logfile = create_log_df('/home/gurleenkaur/Programming/log_analysis/data/logs.log')
+print(df_logfile)
```
Comment (madhavhugar) [inline log.py:73]: Maybe you missed to commit the answers to the questions? :)

**Review by madhavhugar — CHANGES_REQUESTED** [review]

File: log.py (hunk 2 above)
Comment (madhavhugar) [reply log.py:73]: Or did you create a new pull request for the answers?

File: log.py [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+# how many queries, mutations and subscriptions have been performed
+print(df_logfile.type.value_counts().query)
+print(df_logfile.type.value_counts().mutation)
+print(df_logfile.type.value_counts().subscription)
+
+#counts for different operations
+print(df_logfile['name'].value_counts())
+
+#average duration type grouped by operation type
+group_operationtype1 = df_logfile.groupby("type")
```
Comment (ranjanikrishnan) [inline log.py]: Rename to `group_operation_type` :)

File: log.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+mean_operationtype = group_operationtype1.mean()
```
Comment (ranjanikrishnan) [inline log.py]: Here as well, ideally try to follow the naming conventions like [here](https://realpython.com/python-pep8/),

rename to `mean_operation_type` :)

File: log.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+
+df_logfile = create_log_df('/home/gurleenkaur/Programming/log_analysis/data/logs.log')
+print(df_logfile)
+
+
+
+print(df_logfile['name'].nunique())
+# how many queries, mutations and subscriptions have been performed
+print(df_logfile.type.value_counts().query)
```
Comment (ranjanikrishnan) [inline log.py]: You could assign `df_logfile.type.value_counts()` to a variable and reuse that in the later lines like:

```
log_file_value_counts = df_logfile.type.value_counts()
print(f'Queries: {log_file_value_counts.query}')
print(f'Mutation: {log_file_value_counts.mutation}')
print(f'Subscription: {log_file_value_counts.subscription}')
```

Note: the print above uses an improved string formatting which is considered a best practice:)
check f-Strings [here](https://realpython.com/python-f-strings/)

**Review by madhavhugar — CHANGES_REQUESTED** [review]
don't forget to add a README.md file

**Comment by madhavhugar** [comment]
`pip3 freeze > requirements.txt` run this in the project directory when your virtualenv is activated

**Pull request merged**